fastjet/internal/predicates/orientation.go

```diff
@@ -91,3 +104,26 @@ func robustOrientation(x1, y1, x2, y2, x, y *big.Rat) OrientationKind {
 		return Colinear
 	}
 }
+
+func matOrientation(x1, y1, x2, y2, x, y float64) OrientationKind {
+	if (x1 == x2 && x2 == x) || (y1 == y2 && y2 == y) {
+		// points are horizontally or vertically aligned
+		return Colinear
+	}
+	m := mat.NewDense(3, 3, []float64{x1, y1, 1, x2, y2, 1, x, y, 1})
+	logDet, sign := mat.LogDet(m)
+	// FIXME determine maxErrorLog
+	// maxErrorLog is the highest possible number that could lead to a wrong sign due to rounding error
+	maxErrorLog := -10.0
+	if logDet < maxErrorLog {
```

Comment:
I am not quite sure of the logic here.
`LogDet` should return `math.Inf(-1)` if the determinant is 0.

if we've convinced ourselves that `gonum/mat` is robust enough, then we should just test whether `math.IsInf(logDet,-1)`, return `Colinear` and be done with it.

I guess what I am aiming at is to find edge cases where `matOrientation` (written as I suggest above) gives the wrong answer compared to the version using `math/big`)